src/CsvHelper/Configuration/ConfigurationFunctions.cs

```diff
@@ -105,12 +105,12 @@ public static bool ShouldQuote(ShouldQuoteArgs args)
 
 			var shouldQuote = !string.IsNullOrEmpty(args.Field) &&
 			(
-				args.Field.Contains(config.Quote) // Contains quote
-				|| args.Field[0] == ' ' // Starts with a space
-				|| args.Field[args.Field.Length - 1] == ' ' // Ends with a space
-				|| (config.Delimiter.Length > 0 && args.Field.Contains(config.Delimiter)) // Contains delimiter
+				args.Field[0] == ' ' // Starts with a space
```

Comment:
Don't you think it would be better for reading if you replaced ```args.Field[0] == ' '``` with StartWithSpace() internal function? The same with other lines (```args.Field[^1] == ' '``` -> ```EndsWithSpace()```, ```config.Delimiter.Length > 0 && args.Field.Contains(config.Delimiter)``` -> ```ContainsDelimiter()```). And if you do this, you can also remove comments.